exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/ResultRow.kt

```diff
@@ -73,7 +73,18 @@ class ResultRow(
             return c.restoreValueFromParts(rawParts)
         }
 
-        val index = fieldIndex[c]
+        val index = getExpressionIndex(c)
+        return data[index] as T?
+    }
+
+    /**
+     * Retrieves the index of a given expression in the [fieldIndex] map.
+     *
+     * @param c expression for which to get the index
+     * @throws IllegalStateException if expression is not in record set
+     */
+    private fun <T> getExpressionIndex(c: Expression<T>): Int {
+        return fieldIndex[c]
             ?: ((c as? Column<*>)?.columnType as? EntityIDColumnType<*>)?.let { fieldIndex[it.idColumn] }
```

Comment:
I would break down the chain into sequence of lines and early returns, so it will be easier to read

Reply:
I realized this can be further simplified by omitting the second check altogether because it's already covered by the first condition in the `when` block.